Description: Make xlock optional
 The user might prefer a different screen locking application, such
 as xscreensaver or gnome-screensaver.
 .
 Moreover, xlock is not available in Debian.
Author: Andrea Bolognani <eof@kiyuko.org>
Last-Update: 2014-08-19
Index: spectrwm.git/spectrwm.c
===================================================================
--- spectrwm.git.orig/spectrwm.c	2014-08-19 20:38:00.700038567 +0200
+++ spectrwm.git/spectrwm.c	2014-08-19 20:41:58.044050053 +0200
@@ -7030,8 +7030,6 @@
 void
 setup_spawn(void)
 {
-	setconfspawn("lock",		"xlock",		0);
-
 	setconfspawn("term",		"xterm",		0);
 	setconfspawn("spawn_term",	"xterm",		0);
 
@@ -7062,6 +7060,7 @@
 					" -sf $bar_color",	0);
 
 	 /* These are not verified for existence, even with a binding set. */
+	setconfspawn("lock",		"xlock",		SWM_SPAWN_OPTIONAL);
 	setconfspawn("screenshot_all",	"screenshot.sh full",	SWM_SPAWN_OPTIONAL);
 	setconfspawn("screenshot_wind",	"screenshot.sh window",	SWM_SPAWN_OPTIONAL);
 	setconfspawn("initscr",		"initscreen.sh",	SWM_SPAWN_OPTIONAL);
Index: spectrwm.git/spectrwm.1
===================================================================
--- spectrwm.git.orig/spectrwm.1	2014-08-19 20:41:37.660049068 +0200
+++ spectrwm.git/spectrwm.1	2014-08-19 20:41:58.044050053 +0200
@@ -512,16 +512,16 @@
 by freeing the respective key binding.
 .Pp
 For example, to override
-.Ic lock :
+.Ic term :
 .Bd -literal -offset indent
-program[lock] = xscreensaver\-command \-\-lock
+program[term] = rxvt
 .Ed
 .Pp
 To unbind
-.Ic lock
+.Ic term
 and prevent it from being validated:
 .Bd -literal -offset indent
-bind[] = MOD+Shift+Delete
+bind[] = MOD+Shift+Enter
 .Ed
 .Sh BINDINGS
 .Nm
Index: spectrwm.git/spectrwm.conf
===================================================================
--- spectrwm.git.orig/spectrwm.conf	2014-08-19 20:38:00.700038567 +0200
+++ spectrwm.git/spectrwm.conf	2014-08-19 20:42:26.576051436 +0200
@@ -82,19 +82,18 @@
 # PROGRAMS
 
 # Validated default programs:
-# program[lock]		= xlock
 # program[term]		= xterm
 # program[menu]		= dmenu_run $dmenu_bottom -fn $bar_font -nb $bar_color -nf $bar_font_color -sb $bar_border -sf $bar_color
 
 # To disable validation of the above, free the respective binding(s):
-# bind[]		= MOD+Shift+Delete	# disable lock
-# bind[]		= MOD+Shift+Enter	# disable term
+# bind[]		= MOD+Shift+Return	# disable term
 # bind[]		= MOD+p			# disable menu
 
 # Optional default programs that will only be validated if you override:
+# program[lock]			= xlock			# optional
 # program[screenshot_all]	= screenshot.sh full	# optional
 # program[screenshot_wind]	= screenshot.sh window	# optional
-# program[initscr]	= initscreen.sh			# optional
+# program[initscr]		= initscreen.sh		# optional
 
 # EXAMPLE: Define 'firefox' action and bind to key.
 # program[firefox]	= firefox http://spectrwm.org/
